pkg/config/remote/uptane/client.go

```diff
@@ -85,32 +89,85 @@ func (c *Client) Update(response *pbgo.LatestConfigsResponse) error {
 	return c.verify()
 }
 
+// ConfigRootVersion returns the version of the config root.json file
+func (s *State) ConfigRootVersion() uint64 {
+	meta, found := s.ConfigState[metaRoot]
+	if !found {
+		return 0
+	}
+	return meta.Version
+}
+
+// ConfigSnapshotVersion returns the version of the config snapshot.json file
+func (s *State) ConfigSnapshotVersion() uint64 {
+	meta, found := s.ConfigState[metaSnapshot]
+	if !found {
+		return 0
+	}
+	return meta.Version
+}
+
+// DirectorRootVersion returns the version of the director root.json file
+func (s *State) DirectorRootVersion() uint64 {
+	meta, found := s.DirectorState[metaRoot]
+	if !found {
+		return 0
+	}
+	return meta.Version
+}
+
+// DirectorTargetsVersion returns the version of the director targets.json file
+func (s *State) DirectorTargetsVersion() uint64 {
+	meta, found := s.DirectorState[metaTargets]
+	if !found {
+		return 0
+	}
+	return meta.Version
+}
+
 // State returns the state of the uptane client
 func (c *Client) State() (State, error) {
-	c.Lock()
-	defer c.Unlock()
-	configRootVersion, err := c.configLocalStore.GetMetaVersion(metaRoot)
+	targets, err := c.Targets()
 	if err != nil {
 		return State{}, err
 	}
-	directorRootVersion, err := c.directorLocalStore.GetMetaVersion(metaRoot)
-	if err != nil {
-		return State{}, err
+
+	c.Lock()
+	defer c.Unlock()
+
+	s := State{
+		ConfigState:   map[string]MetaState{},
+		DirectorState: map[string]MetaState{},
 	}
-	configSnapshotVersion, err := c.configLocalStore.GetMetaVersion(metaSnapshot)
+
+	metas, err := c.configLocalStore.GetMeta()
 	if err != nil {
 		return State{}, err
 	}
-	directorTargetsVersion, err := c.directorLocalStore.GetMetaVersion(metaTargets)
+
+	for metaName, content := range metas {
+		version, err := metaVersion(content)
+		if err == nil {
+			s.ConfigState[metaName] = MetaState{Version: version, Hash: metaHash(content)}
+		}
+	}
+
+	directorMetas, err := c.directorLocalStore.GetMeta()
 	if err != nil {
 		return State{}, err
 	}
-	return State{
-		ConfigRootVersion:      configRootVersion,
-		ConfigSnapshotVersion:  configSnapshotVersion,
-		DirectorRootVersion:    directorRootVersion,
-		DirectorTargetsVersion: directorTargetsVersion,
-	}, nil
+
+	for metaName, content := range directorMetas {
+		version, err := metaVersion(content)
+		if err == nil {
+			s.DirectorState[metaName] = MetaState{Version: version, Hash: metaHash(content)}
+		}
+	}
+	for targetName := range targets {
+		s.DirectorState[targetName] = MetaState{Version: 0, Hash: ""}
+	}
```

Comment:
I've updated the State structure to have target files as `[]string` ; since I'm not sure their hash is useful, correct me if I'm wrong but the useful information here is the target path. What do you think?

Reply:
Oh darn, yeah I meant "path"!

As far as usefulness, I suppose it depends on how this primarily gets used. It's probably not all that useful to the end customers. However, I could see the hash of the target files being useful if we're troubleshooting an issue. Maybe a verbose flag for the command that would print the hash as well so it's not too noisy by default?